apps/hash-api/src/graph/knowledge/primitive/entity/after-update-entity-hooks/file-document-after-update-entity-hook-callback.ts

```diff
@@ -0,0 +1,92 @@
+import { VersionedUrl } from "@blockprotocol/type-system";
+import { getWebMachineActorId } from "@local/hash-backend-utils/machine-actors";
+import { systemEntityTypes } from "@local/hash-isomorphic-utils/ontology-type-ids";
+import type { ParseTextFromFileParams } from "@local/hash-isomorphic-utils/parse-text-from-file-types";
+import { simplifyProperties } from "@local/hash-isomorphic-utils/simplify-properties";
+import type { DOCXDocumentProperties } from "@local/hash-isomorphic-utils/system-types/docxdocument";
+import type { PDFDocumentProperties } from "@local/hash-isomorphic-utils/system-types/pdfdocument";
+import type { PPTXPresentationProperties } from "@local/hash-isomorphic-utils/system-types/pptxpresentation";
+import { Entity, extractOwnedByIdFromEntityId } from "@local/hash-subgraph";
+
+import { isStorageType, storageProviderLookup } from "../../../../../storage";
+import { UpdateEntityHookCallback } from "../update-entity-hooks";
+
+export const entityTypesToParseTextFrom: VersionedUrl[] = [
+  systemEntityTypes.docxDocument.entityTypeId,
+  systemEntityTypes.pdfDocument.entityTypeId,
+  systemEntityTypes.pptxPresentation.entityTypeId,
+];
+
+type FileEntityToParseProperties =
+  | DOCXDocumentProperties
+  | PDFDocumentProperties
+  | PPTXPresentationProperties;
+
+export const parseTextFromFileAfterUpdateEntityHookCallback: UpdateEntityHookCallback =
+  async ({ entity, updatedProperties, context, authentication }) => {
+    const { temporalClient } = context;
+
+    const docxFileEntity =
```

Comment:
Should this be called `docx` when it might be two other types?

Reply:
Addressed in https://github.com/hashintel/hash/pull/3929/commits/94af95d3854c8caba8847b1363a76af55f46de5e